pkg/daemon/update.go

```diff
@@ -2848,3 +2920,82 @@ func (dn *Daemon) hasImageRegistryDrainOverrideConfigMap() (bool, error) {
 
 	return false, fmt.Errorf("Error fetching image registry drain override configmap: %w", err)
 }
+
+// Enables the revert layering systemd unit.
+//
+// This requires the current MachineConfig because it has the original unit
+// that we use to create the cloned unit.
+//
+// To enable the unit, it performs the following operations:
+// 1. It gets the disabled revert systemd unit from the MachineConfig, renames
+// it, enables it, and unmasks it.
+// 2. It writes the new systemd unit to disk using the default MCD code paths
+// for that process.
+func (dn *Daemon) enableRevertSystemdUnit(mc *mcfgv1.MachineConfig) error {
+	clonedUnit, err := getRevertSystemdUnitFromMachineConfig(mc)
```

Comment:
Question: instead of having a machineconfig template that mostly sits idle (and only would take effect by getting cloned to a different name), would it make sense to just create that unit completely on the fly here in golang? And then the removal post-revert would just remove the unit.

This way we don't have an additional unit sitting in `templates/common/_base/units`, or alternatively, we still use a template, just not dropped to the disk directly via that, and we apply it directly via our `manifest` folder or something

Reply:
> would it make sense to just create that unit completely on the fly here in golang? And then the removal post-revert would just remove the unit.

I was thinking about that actually. You gave me some good ideas for things to try!

Reply:
For posterity, I settled on creating a subpackage called `pkg/daemon/runtimeassets` to house this template as well as the code needed to render it and all required tests. I briefly considered using https://github.com/coreos/go-systemd/tree/main/unit to programmatically construct the systemd unit to avoid rendering templates, but ultimately determined that it was less maintainable.

I added some additional context in a README file for this subpackage's intended purpose and use.